tonyear/core.py

```diff
@@ -65,24 +65,65 @@ def get_discounted_curve(discount_rate: float, curve: np.ndarray) -> np.ndarray:
 
 
 def print_benefit_report(method_output: dict) -> None:
-    '''print the benefit report'''
-    discount = str(round(method_output['parameters']['discount_rate'] * 100, 1))
-    delay = str(method_output['parameters']['delay'])
-    baseline_atm_cost = str(round(method_output['baseline_atm_cost'], 2))
-    benefit = str(round(method_output['benefit'], 2))
-    num_needed = str(round(method_output['num_for_equivalence'], 1))
+    """print the benefit report"""
+    discount = str(round(method_output["parameters"]["discount_rate"] * 100, 1))
+    delay = str(method_output["parameters"]["delay"])
+    baseline_atm_cost = str(round(method_output["baseline_atm_cost"], 2))
+    benefit = str(round(method_output["benefit"], 2))
+    num_needed = str(round(method_output["num_for_equivalence"], 1))
 
     print()
-    print('Discount rate: ' + discount + '%')
-    print('Delay: ' + delay + ' year(s)')
-    print('Baseline atmospheric cost: ' + baseline_atm_cost + ' ton-years')
-    print('Benefit from 1tCO2 with delay: ' + benefit + ' ton-years')
-    print('Number needed: ' + num_needed)
+    print("Discount rate: " + discount + "%")
+    print("Delay: " + delay + " year(s)")
+    print("Baseline atmospheric cost: " + baseline_atm_cost + " ton-years")
+    print("Benefit from 1tCO2 with delay: " + benefit + " ton-years")
+    print("Number needed: " + num_needed)
     print()
 
 
+def get_avoided_comparison(
+    tonyear_calc: dict,
+    baseline_curve: np.array,
+    integration_time: int,
+) -> dict:
+    """
+    Compare cumulative radiative forcing of delaying 1 tCO2  against cumulative radiative forcing of permanently
+     avoiding the emission of a smaller amount of CO2.
+    Parameters
+    ----------
+    tonyear_calc: dict
+        output from `calculate_tonyear`
+    baseline_curve: np.array
+        CO2 IRF curve
+    integration_time: int
+        time period to evaluate equivalence claim; does _not_ have to equal time_horizon from IRF curve generation
```

Comment:
Is there a good reason not to take "baseline_curve" and "integration_time" from "tonyear_calc" (baseline and time_horizon, respectively)?

Reply:
Maybe. How does the following logic sound to you? 

- baseline curve: My gut tells me that we always want to do equivalency comparisons against an undiscounted baseline curve. `calculate_tonyear`, however, returns the baseline curve with discounting applied. 
- integration time: here, again, I think there is actually an interesting use case for manually specifying integration time. in `calculate_tonyear` you're making the assertion that 'all i care about is the next `time_horizon` years'. When evaluating equivalency claims, you might want to explore i) those equivalency claims over the period `time_horizon` or ii) over time periods that better match the lifetime of CO2 emissions (1000+ years)

Reply:
Those are both compelling to me!